compiler/GHC/Linker/Dynamic.hs

```diff
@@ -105,7 +106,8 @@ linkDynLib logger tmpfs dflags0 unit_env o_files dep_packages
     pkg_framework_opts <- getUnitFrameworkOpts unit_env (map unitId pkgs)
     let framework_opts = getFrameworkOpts (initFrameworkOpts dflags) platform
 
-    let linker_config = initLinkerConfig dflags
+    let require_cxx = any ((==) (PackageName (fsLit "system-cxx-std-lib")) . unitPackageName) pkgs
```

Comment:
I think we should be ignoring `system-cxx-std-lib` outright, and just look at the `-l` flags we pass to the linker. If that includes `-lc++`, or `-lstdc++`, or similar, we just swap out `cc` for `cxx`. Ultimately the `system-cxx-std-lib` package should be removed outright and not exist. 

cabal packages that need to link c++ will have `extra-libraries: c++` or similar anyway in their dependency tree. And if they don't because they now depend on this `system-cxx-std-lib`, upstream has just made the issue worse :facepalm:

Reply:
In the upstream ticket (https://gitlab.haskell.org/ghc/ghc/-/issues/25767) Ben made the argument that we can't really use `cxx` to link as what would happen if we needed to link with another language too (e.g. rust)? So upstream seems keen to to continue to link with `ld/cc`.

Not sure what to do then. Should `stable-haskell`'s GHC diverge from upstream here?

Reply:
Just echoing what I wrote on the upstream issue: I *do* think `system-cxx-std-lib` is a good solution *provided* that it is one in "user space" (`cabal`, `ghc-toolchain`), not shoved into the GHC build system as it is today.